lambda-buffers-compiler/src/LambdaBuffers/Compiler/TypeClass/Rules.hs

```diff
@@ -0,0 +1,49 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE TypeFamilies #-}
+
+module LambdaBuffers.Compiler.TypeClass.Rules (
+  ClassRef (..),
+  Class (..),
+  Constraint (..),
+  Rule (..),
+  type Instance,
+  mapPat,
+) where
+
+import LambdaBuffers.Compiler.ProtoCompat qualified as P
+import LambdaBuffers.Compiler.TypeClass.Pat (Pat)
+
+data ClassRef = CRef {cname :: P.ClassName, cmodule :: P.ModuleName}
+  deriving stock (Show, Eq, Ord)
+
+data Class = Class
+  { name :: ClassRef
```

Comment:
This is wrong, ClassRef is a reference, this should be a ClassName?

Reply:
It's not wrong exactly, though `ClassRef` is probably a bad name for that type (forgive me I'm running out of good names). `ClassRef` == `FullyQualifiedClassName` because we need to know which module the class is defined in.